### ISSUE TITLE
Cops restricted by `Include` are not run when linting a single file

### ISSUE THREAD

**Issue by Earlopain** [rubocop/rubocop#13124]
## Expected behavior

If I pass a cop to run with `--only` it should run, even if it restricts itself to certain folders/patterns through the `Include` config. 

The repro below is a bit contrived since not many buildin cops use `Include` but the same is not true for extensions like `rubocop-rails`.

## Steps to reproduce the problem

```yml
# .rubocop.yml

AllCops:
  DisabledByDefault: enable

Style/StringLiterals:
  Enabled: true
  Include:
    - lib/**/*.rb
```

```rb
# test.rb

""
''
```

```
earlopain@DESKTOP-PC rubocop]$ bundle exec rubocop test.rb --only Style/StringLiterals
Inspecting 1 file
.

1 file inspected, no offenses detected
```

Moving the file into the lib folder, or removing the `Include` config makes the cop run.

## RuboCop version

```
$ bundle exec rubocop -V
1.65.1 (using Parser 3.3.4.2, rubocop-ast 1.32.0, running on ruby 3.4.0) [x86_64-linux]
```

**Comment by Earlopain** [comment]
This would also apply to `Exclude`.

I'm going to close this since I'm doing something unconventional and a solution would benefit barely anyone (if any).

Additional, this is only one case where you can apply this logic to. For example, if running rubocop from the app root against another  application in a folder like `../my_other_app`. It's  not possible for rubocop to correctly infer user intention.
